src/app/services/xml-parsers/header-parser.ts

```diff
@@ -0,0 +1,218 @@
+import { isNestedInElem } from 'src/app/utils/dom-utils';
+import {
+  EditionStmt, Extent, FileDesc, GenericElement, MsDesc, NamedEntityRef, Note,
+  NotesStmt, PublicationStmt, Resp, RespStmt, SeriesStmt, SourceDesc, TitleStmt, XMLElement,
+} from '../../models/evt-models';
+import { AttrParser, GenericParser, NoteParser, queryAndParseElement, queryAndParseElements } from './basic-parsers';
+import { MsDescParser } from './msdesc-parser';
+import { NamedEntityRefParser } from './named-entity-parsers';
+import { createParser, getClass, parseChildren, Parser } from './parser-models';
+
+export class RespParser extends AttrParser implements Parser<XMLElement> {
+  parse(xml: XMLElement): Resp {
+    const attributes = this.attributeParser.parse(xml);
+    const { ref, when } = attributes;
+    const normalizedResp = ref?.indexOf('http://') < 0 && ref?.indexOf('https://') < 0 ? `http://${ref}` : ref ?? '';
+
+    return {
+      type: Resp,
+      class: getClass(xml),
+      content: parseChildren(xml, this.genericParse, true),
+      attributes,
+      normalizedResp,
+      date: when || '',
+    };
+  }
+}
+
+export class RespStmtParser extends AttrParser implements Parser<XMLElement> {
+  private noteParser = createParser(NoteParser, this.genericParse);
+  private respParser = createParser(RespParser, this.genericParse);
+  private namedEntityRefParser = createParser(NamedEntityRefParser, this.genericParse);
+
+  parse(xml: XMLElement): RespStmt {
+    const people = Array.from(xml.querySelectorAll<XMLElement>(':scope > name, :scope > orgName, :scope > persName'))
+      .map(p => {
+        if (p.tagName === 'orgName' || p.tagName === 'persName') {
+          return this.namedEntityRefParser.parse(p) as NamedEntityRef;
+        }
+
+        return this.genericParse(p) as GenericElement;
+      });
+
+    return {
+      type: RespStmt,
+      class: getClass(xml),
+      content: parseChildren(xml, this.genericParse, true),
+      attributes: this.attributeParser.parse(xml),
+      responsibility: queryAndParseElement<Resp>(xml, 'resp', this.respParser),
+      notes: queryAndParseElements<Note>(xml, 'note', this.noteParser),
+      people,
+    };
+  }
+}
+
+export class TitleStmtParser extends GenericParser implements Parser<XMLElement> {
+  private respStmtParser = createParser(RespStmtParser, this.genericParse);
+
+  parse(xml: XMLElement): TitleStmt {
+    const title = queryAndParseElements<GenericElement>(xml, 'title[type="main"]', this.genericElemParser);
+
+    return {
+      type: TitleStmt,
+      class: getClass(xml),
+      content: parseChildren(xml, this.genericParse),
+      attributes: this.attributeParser.parse(xml),
```

Comment:
you can leverage the GenericParser for these properties (look at manuscript parsers as an example)